### PR TITLE
Create color palettes for Periodic Table of elements

### PR DESCRIPTION
**Pull request by MarianoGnu**
-----------
Part 1 of Task: Change the Colors of the Academic View to the CPK colors
Related to #34

### CODE REVIEW

File: godot_project/autoloads/data/color_palettes/periodic_table_color_palette.gd [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+func _get_property_list() -> Array[Dictionary]:
+	var properties: Array[Dictionary] = []
+	properties.append({
+		"name": "page",
+		"type": TYPE_INT,
+		"hint": PROPERTY_HINT_RANGE,
+		"hint_string": "1,12,1"
+	})
+	_page = clamp(_page, 1, 12)
+	for i in range((_page-1)*10, (_page*10), 1):
```
Comment (kubecz3k) [inline godot_project/autoloads/data/color_palettes/periodic_table_color_palette.gd:45]: it seems like we really could use some constants like "NMB_OF_ITEMS_ON_PAGE / MAX_NMB_OF_PAGES / etc" and others to both better self document those numbers and what they are representing as well as to ease the maintenance/changes for people who are not familiar with how this logic works

File: godot_project/autoloads/data/color_palettes/periodic_table_color_palette.gd [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+			"type": TYPE_BOOL,
+			"usage": PROPERTY_USAGE_DEFAULT, # See above assignment.
+			"hint": PROPERTY_HINT_NONE,
+			"hint_string": ""
+		}
+	return properties
+
+func _set(in_property: StringName, in_value: Variant) -> bool:
+	if in_property == &"page":
+		if _page != clamp(in_value, 1, 12):
```
Comment (kubecz3k) [inline godot_project/autoloads/data/color_palettes/periodic_table_color_palette.gd:97]: constants request applies here as well

File: godot_project/autoloads/data/color_palettes/periodic_table_color_palette.gd [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+	if Engine.is_editor_hint() and subproperty == "set" and in_value == false:
+		var element_data: Dictionary = _data[atomic_number]
+		for to_erase: String in ["color","noise_color","bond_color","font_color"]:
+			element_data.erase(to_erase)
+	if Engine.is_editor_hint() and subproperty == "color":
+		var color: Color = in_value as Color
+		var luminisence: float = (color.r + color.g + color.b) / 3.0
+		if not _data[atomic_number].has("noise_color"):
+			var noise_color: Color
+			if luminisence > 0.5:
```
Comment (kubecz3k) [inline godot_project/autoloads/data/color_palettes/periodic_table_color_palette.gd:119]: those numbers also could use some constants, I can see various values all over this function, would be good to be sure those are not the same by a chance

**Review by kubecz3k — APPROVED** [review]
would be nice to improve the situation with magic numbers before the merge, otherwise :+1:

**Comment by MarianoGnu** [comment]
> would be nice to improve the situation with magic numbers before the merge, otherwise 👍

Given there's another PR depending on this one will be easier to make a followup PR later 👍

**Comment by MarianoGnu** [comment]
Created https://github.com/MSEP-one/msep.one/pull/49 to followup feedback

**Pull request merged**